### PR TITLE
feat(kubernetes): create complex k8s vertices

### PR DESCRIPTION
**Pull request by achiar99**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
Extract k8s pods resources and create vertices with the metadata for edges

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by tronxd — APPROVED** [review]

File: checkov/kubernetes/graph_builder/local_graph.py [hunk 1]
```diff
@@ -1,51 +1,92 @@
...
+                resource_type = resource.get('kind', DEFAULT_RESOURCE_TYPE)
                 metadata = resource.get('metadata') or {}
                 # TODO: add support for generateName
-                if is_invalid_k8_definition(resource) or not metadata.get('name'):
-                    logging.info(f"failed to create a vertex in file {file_path}")
-                    file_conf.remove(resource)
-                    continue
+                
+                if resource_type == DEFAULT_RESOURCE_TYPE:
+                    if not is_invalid_k8_pod_definition(resource):
```
Comment (PelegLi) [inline checkov/kubernetes/graph_builder/local_graph.py:34]: Should the `not` be in the condition?

File: checkov/kubernetes/graph_builder/local_graph.py [hunk 2]
```diff
@@ -1,51 +1,104 @@
...
                 if resource.get('kind') == "List":
                     file_conf.extend(item for item in resource.get("items", []) if item)
                     file_conf.remove(resource)
 
+            if create_complex_vertices:
+                file_conf = self._extract_nested_resources(file_conf)
+
             for resource in file_conf:
-                resource_type = resource.get('kind')
+                resource_type = resource.get('kind', DEFAULT_RESOURCE_TYPE)
```
Comment (PelegLi) [inline checkov/kubernetes/graph_builder/local_graph.py:29]:
```suggestion
                resource_type = resource.get('kind', DEFAULT_NESTED_RESOURCE_TYPE)
```

File: checkov/kubernetes/graph_builder/local_graph.py [hunk 3]
```diff
@@ -1,51 +1,104 @@
 import logging
 from copy import deepcopy
-from typing import List, Dict
+from typing import Any, List, Dict
 
 from checkov.common.graph.graph_builder.local_graph import LocalGraph
-from checkov.kubernetes.graph_builder.graph_components.blocks import KubernetesBlock
-from checkov.kubernetes.kubernetes_utils import is_invalid_k8_definition, get_resource_id
+from checkov.kubernetes.graph_builder.graph_components.blocks import KubernetesBlock, KubernetesBlockMetadata, KubernetesSelector
+from checkov.kubernetes.kubernetes_utils import DEFAULT_RESOURCE_TYPE, is_invalid_k8_definition, get_resource_id, is_invalid_k8_pod_definition
```
Comment (PelegLi) [inline checkov/kubernetes/graph_builder/local_graph.py:7]:
```suggestion
from checkov.kubernetes.kubernetes_utils import DEFAULT_NESTED_RESOURCE_TYPE, is_invalid_k8_definition, get_resource_id, is_invalid_k8_pod_definition
```

File: checkov/kubernetes/kubernetes_utils.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 from checkov.kubernetes.parser.parser import parse
 
 K8_POSSIBLE_ENDINGS = {".yaml", ".yml", ".json"}
+DEFAULT_RESOURCE_TYPE = 'Pod'
```
Comment (PelegLi) [inline checkov/kubernetes/kubernetes_utils.py:20]:
```suggestion
DEFAULT_NESTED_RESOURCE_TYPE = 'Pod'
```

File: checkov/kubernetes/graph_builder/local_graph.py [hunk 5]
```diff
@@ -1,51 +1,104 @@
...
-                resource_type = resource.get('kind')
+                resource_type = resource.get('kind', DEFAULT_RESOURCE_TYPE)
                 metadata = resource.get('metadata') or {}
                 # TODO: add support for generateName
-                if is_invalid_k8_definition(resource) or not metadata.get('name'):
-                    logging.info(f"failed to create a vertex in file {file_path}")
-                    file_conf.remove(resource)
-                    continue
+                
+                if resource_type == DEFAULT_RESOURCE_TYPE:
```
Comment (PelegLi) [inline checkov/kubernetes/graph_builder/local_graph.py:33]:
```suggestion
                if resource_type == DEFAULT_NESTED_RESOURCE_TYPE:
```

File: checkov/kubernetes/kubernetes_utils.py [hunk 6]
```diff
@@ -177,13 +178,34 @@ def is_invalid_k8_definition(definition: Dict[str, Any]) -> bool:
...
+    labels = metadata.get('labels')
+    name = metadata.get('name')
+    if name is None and labels is None:
+        return False
+    return True
 
 
 def get_resource_id(resource: Dict[str, Any]) -> Optional[str]:
-    resource_type = resource["kind"]
+    resource_type = resource.get("kind", DEFAULT_RESOURCE_TYPE)
```
Comment (PelegLi) [inline checkov/kubernetes/kubernetes_utils.py:200]:
```suggestion
    resource_type = resource.get("kind", DEFAULT_NESTED_RESOURCE_TYPE)
```

**Review by PelegLi — APPROVED** [review]
Great!

**Pull request merged**